lib/plugins/helper/js.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 function jsHelper(...args) {
-  return args.reduce((result, path, i) => {
-    if (i) result += '\n';
+  let result = '\n';
+  let items = args;
 
-    if (Array.isArray(path)) {
-      return result + Reflect.apply(jsHelper, this, path);
+  if (!Array.isArray(args)) {
+    items = [args];
+  }
+
+  items = items.flat();
+
+  items.forEach(item => {
+    // Old syntax
+    if (typeof item === 'string' || item instanceof String) {
+      let path = item;
+      if (!path.endsWith('.js')) {
+        path += '.js';
+      }
+      result += `<script src="${this.url_for(path)}"></script>\n`;
+    } else {
+      // New syntax
+      let tmpResult = '<script';
+      for (const attribute in item) {
+        if (attribute === 'src' && !item[attribute].endsWith('.js')) {
+          item[attribute] += '.js';
+        }
+        if (item[attribute] === true) tmpResult += ' ' + attribute;
+        else tmpResult += ` ${attribute}="${item[attribute]}"`;
+      }
```

Comment:
In the current hexo code, the path is defined as `this.url_for(path)`.  
We should do the same for the src attribute. 
ex: 
``` 
if (attribute === 'src') {
item[attribute] = this.url_for(item[attribute])
}
```

Reply:
Good catch. Added the `url_for()`  and test case.